packages/common-grpc/src/service.js

```diff
@@ -372,12 +373,14 @@ GrpcService.prototype.requestStream = function(protoOpts, reqOpts) {
     return stream;
   }
 
-  var retryOpts = {
+  var retryOpts = extend({
     retries: this.maxRetries,
+    currentRetryAttempt: 0,
     objectMode: objectMode,
     shouldRetryFn: GrpcService.shouldRetryRequest_,
 
     request: function() {
+      setImmediate(() => stream.emit('request'));
```

Comment:
We should actually do this from `retry-request`, since any use of that library would be interested in knowing when each request is being retried. What do you think?

Reply:
Agreed, should I create a PR there to emit that in `retry-request` when `streamMode` is `true`?

Reply:
That would be great, thank you.

Reply:
Done stephenplusplus/retry-request#18

Reply:
Thanks! Released under retry-request@3.3.0.

Reply:
Thanks, I'm not exactly sure at this point how to have the bigtable client aware of requests now, what would be the best way to communicate that a request happened? Something like 

```js
      return service[protoOpts.method](reqOpts, grpcMetadata, grpcOpts)
        .on('request', function() {
          setImmediate(function() {
            stream.emit('request')
          })
        })
```

Doesn't call the callback

Reply:
The listener be attached to retryRequest and then forwarded to `stream`:

```js
return retryRequest(null, retryOpts)
  .on('error', function(err) {
    var grpcError = GrpcService.decorateError_(err);
 
    stream.destroy(grpcError || err);
  })
  .on('request', stream.emit.bind(stream, 'request'))
  .pipe(stream);
```

Reply:
Updated the PR